server/forge/github/github.go

```diff
@@ -394,16 +394,29 @@ func (c *client) newConfig(req *http.Request) *oauth2.Config {
 
 // helper function to return the GitHub oauth2 client
 func (c *client) newClientToken(ctx context.Context, token string) *github.Client {
-	ts := oauth2.StaticTokenSource(
-		&oauth2.Token{AccessToken: token},
-	)
-	tc := oauth2.NewClient(ctx, ts)
-	if c.SkipVerify {
-		tc.Transport.(*oauth2.Transport).Base = &http.Transport{
-			Proxy: http.ProxyFromEnvironment,
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify: true,
-			},
+	var tc *http.Client
+	if token != "" {
```

Comment:
Why could the token be empty?

Reply:
I think that could be an issue if the repo is private and the woodpecker project is public. Could we pass the token of the project / repo instead?